app/Controllers/QueryController.js

```diff
@@ -0,0 +1,18 @@
+const { getQueryVocabulary } = require('../Services/QueryServiceProvider.js');
+const { parseTokenUserId } = require('../utils/index.js');
+
+async function sendQueryVocabulary(req, res) {
+  // get userId from request
+  const userId = await parseTokenUserId(req);
+  const languagePackageId = req.params.languagePackageId;
+  const limit = req.query.limit;
```

Comment:
could you use destructuring?
```js
const { languagePackageId, limit } = req.query;
```

Reply:
can't be used as languagePackageId ist a param and limit a query value

Reply:
oh sorry, I overread that. Than use:
```js
const { languagePackageId } = req.params;
const { limit } = req.query;
```
I think it looks a bit better.